File: libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/calls/TurnServerInfo.java
package org.whispersystems.signalservice.api.messages.calls;


import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class TurnServerInfo {

  @JsonProperty
  private String username;

  @JsonProperty
  private String password;

  @JsonProperty
  private String hostname;

  @JsonProperty
  private List<String> urls;

  @JsonProperty
  private List<String> urlsWithIps;

  public String getUsername() {
    return username;
  }

  public String getPassword() {
    return password;
  }

  // Hostname for the ips in urlsWithIps
  public String getHostname() {
    return hostname;
  }

  public List<String> getUrls() {
    return urls;
  }

  public List<String> getUrlsWithIps() {
    return urlsWithIps;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/calls/TurnServerInfo.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.messages.calls

data class TurnServerInfo(
  val username: String? = null,
  val password: String? = null,
  /** Hostname for the ips in [urlsWithIps]. */
  val hostname: String? = null,
  val urls: List<String>? = null,
  val urlsWithIps: List<String>? = null,
  val ttl: Long? = null
)
